internal/cmd/update.go

```diff
@@ -0,0 +1,309 @@
+package cmd
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"runtime"
+	"strings"
+	"time"
+
+	"cdr.dev/coder-cli/internal/config"
+	"cdr.dev/coder-cli/internal/version"
+	"cdr.dev/coder-cli/pkg/clog"
+	"golang.org/x/xerrors"
+
+	"github.com/blang/semver/v4"
+	"github.com/manifoldco/promptui"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+// updater updates coder-cli.
+type updater struct {
+	confirmF       func(label string) (string, error)
+	executablePath string
+	fs             afero.Fs
+	httpClient     getter
+	versionF       func() string
+}
+
+func updateCmd() *cobra.Command {
+	var (
+		force    bool
+		coderURL string
+	)
+
+	cmd := &cobra.Command{
+		Use:   "update",
+		Short: "Update coder binary",
+		Long:  "Update coder to the version matching a given coder instance.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			httpClient := &http.Client{
+				Timeout: 10 * time.Second,
+			}
+
+			updater := &updater{
+				httpClient:     httpClient,
+				fs:             afero.NewOsFs(),
+				confirmF:       defaultConfirm,
+				versionF:       func() string { return version.Version },
+				executablePath: os.Args[0],
+			}
+			return updater.Run(ctx, force, coderURL)
+		},
+	}
+
+	cmd.Flags().BoolVar(&force, "force", false, "do not prompt for confirmation")
+	cmd.Flags().StringVar(&coderURL, "coder", "", "coder instance against which to match version")
+
+	return cmd
+}
+
+type getter interface {
+	Get(url string) (*http.Response, error)
+}
+
+func (u *updater) Run(ctx context.Context, force bool, coderURLString string) error {
+	// TODO: check under following directories and warn if coder binary is under them:
+	//   * homebrew prefix
```

Comment:
Should we run `brew update ...` for them if that's the case?

Reply:
I'd find that a bit presumptuous myself. It's probably safer to tell users to run brew update manually.

Reply:
Good point!